org.alloytools.alloy.application/src/main/java/edu/mit/csail/sdg/alloy4viz/VizGUI.java

```diff
@@ -794,15 +883,35 @@ private void updateDisplay() {
         openEvaluatorButton.setVisible(!isMeta && settingsOpen == 0 && evaluator != null);
         closeEvaluatorButton.setVisible(!isMeta && settingsOpen == 2 && evaluator != null);
         enumerateMenu.setEnabled(!isMeta && settingsOpen == 0 && enumerator != null);
-        enumerateButton.setVisible(!isMeta && settingsOpen == 0 && enumerator != null);
+        enumerateMenu.setVisible(!isTrace); // [HASLab]
+        enumerateButton.setVisible(!isMeta && settingsOpen == 0 && enumerator != null && !isTrace); // [HASLab]
+        initMenu.setEnabled(!isMeta && settingsOpen == 0 && enumerator != null); // [HASLab]
+        initMenu.setVisible(isTrace); // [HASLab]
+        initButton.setVisible(!isMeta && settingsOpen == 0 && enumerator != null && isTrace); // [HASLab]
+        pathMenu.setEnabled(!isMeta && settingsOpen == 0 && enumerator != null && !seg_iteration); // [HASLab]
```

Comment:
after a fork in a particular state, cannot ask for a new full path until new config